Scto guidelines (#10)
worldbank/dime-standards: dime-research-standards/pillar-4-data-security/data-security-resources/surveycto-encryption-guidelines.md

So multiple forms in the same project can use the same key,
but the same key should not be used across projects.

As the names suggests it is very important that you keep your private key secret.
If you accidentally publish your private key, then your data is no longer securely protected.
Additionally, if you lose access to your private key, then there is no way to decrypt your data.

There is no security risk of accidentally publishing your public key.
If you were to lose your public key, then you will not be able to encrypt any more data
(until you create a new public/private key pair)
but as long as you have the private key you will still be able to decrypt data encrypted with the lost public key.

### Step 1.1 - Creating a public/private key pair.

You can create the public/private key pair on your SurveyCTO server.
Log in to your server, and navigate to the _Design_ tab. Then click _Tools_.

<img src="https://github.com/worldbank/dime-standards/blob/scto-guidelines/dime-research-standards/pillar-4-data-security/data-security-resources/img/scto-encrypt-create-1.png" width="75%"><!--- Image is read from master branch or use full URL-->

> Then click _create a new key_ and then _Start key generator_

Author: done in efac4632a617c269d6083b8a7ffa1a2ff104ea00

After storing these keys in a password manager,
delete these files from every location it is saved in.
If they keys were already sent to the cloud,
then there is no way to fully delete them.

### Step 1.2 - Securely share and long term store the key pair

Saving the key pair in a regular folder on your computer is not a secure enough way of storing the key files.
Instead, our recommendation is that the key is stored in a password manager.
Make sure that you have a password manager set up and
that you are comfortable using it before proceeding with these instructions.
We will provide instructions for the password manager LastPass,
but this can be done in other password managers too.
A secure alternative to saving to storing the keys in a password manager
is to store the keys in an [encrypted folder](https://github.com/worldbank/dime-standards/blob/scto-guidelines/dime-research-standards/pillar-4-data-security/data-security-resources/veracrypt-guidelines.mdt)
on your computer,
but then you still need to store the key to the encrypted folder in a password manager.

Go to lastpass.com, log in to your vault and
click the plus sign in the red circle to create a new item.

> Select _Secure Note_

Author: done in efac4632a617c269d6083b8a7ffa1a2ff104ea00

<img src="https://github.com/worldbank/dime-standards/blob/scto-guidelines/dime-research-standards/pillar-4-data-security/data-security-resources/img/scto-encrypt-store-1.png" width="50%"><!--- Image is read from master branch or use full URL-->

> Then copy all the content of both keys you created and downloaded from your SurveyCTO server.

Author: done in efac4632a617c269d6083b8a7ffa1a2ff104ea00

> Then copy all the content of both keys you created and downloaded from your SurveyCTO server.

Author: done in e6f8855dd567faf20c5955f41f8ced7e3486abaf

Paste the value of the public key in the cell in the first row of that column.
In this method it is important that you do **not** include
the  key header `-----BEGIN PUBLIC KEY-----`
and the key footer `-----END PUBLIC KEY-----`. See example below.

<img src="https://github.com/worldbank/dime-standards/blob/scto-guidelines/dime-research-standards/pillar-4-data-security/data-security-resources/img/scto-encrypt-b1.png" width="25%"><!--- Image is read from master branch or use full URL-->

Go to the _Design_ tab in your SurveyCTO server. Click _Upload form definition_.

<img src="https://github.com/worldbank/dime-standards/blob/scto-guidelines/dime-research-standards/pillar-4-data-security/data-security-resources/img/scto-encrypt-1.png" width="75%"><!--- Image is read from master branch or use full URL-->

Upload the form you with the public key included in the settings tab,
and then follow the instructions as normal.
SurveyCTO will test that there are no errors in the public key,
but you should never start collecting data using an encrypted form before
you have followed our test instructions below.

## Topic 3 - Publishable fields

> One often overseen feature when encrypting forms are publishable fields.

Author: haha, this is funny! Thanks!

Then you can give the survey firm permission to download data from the server
or view it in the Data Explorer,
but as long as they do not have access to the private key,
they have no access to fields that not publishable.

To make a field publishable, you simply write "_yes_" in the _publishable_ column in the _survey_ tab in the questionnaire form.

To download the publishable data, go to the _Export_ tab in your SurveyCTO server.
Find your form and click "_Download for data_".
Then in the two sections as normal,
but in the "_Fields to include_" make sure that the checkbox
"_Publishable fields only (if you don't have the private key)_" is checked.
Then click _Download .csv now_.

<img src="https://github.com/worldbank/dime-standards/blob/scto-guidelines/dime-research-standards/pillar-4-data-security/data-security-resources/img/scto-publish-1.png" width="75%"><!--- Image is read from master branch or use full URL-->

## Topic 4 - Using your private key to decrypt your data in SurveyCTO

Technically you can download encrypted data from your SurveyCTO server using a browser,
but then you first need to save your private key in a file on your computer.
In the workflow we recommend,

> you really want to avoid to ever store the private key in a file on your computer

Reviewer: ```suggestion
you avoid ever storing the private key in a file on your computer
```

after you have saved it in a password manager.

> Instead, we recommend you to only download data using [SurveyCTO Desktop](https://docs.surveycto.com/desktop/)

Reviewer: ```suggestion
Instead, we recommend to only download data using [SurveyCTO Desktop](https://docs.surveycto.com/desktop/)
```

as you can copy your key from your password manager and paste it
without having to save it in a file first.

In SurveyCTO Desktop you log in to your server,
and click _Sync_ in the menu to the left.
Fill in the information as normal,
and in step 2 you need to paste the private key. See image below.
Go to your password manager and copy your private key.
Remember to include the header `-----BEGIN RSA PRIVATE KEY-----`
and the footer `-----END RSA PRIVATE KEY-----`.
Once you have copied it to your clipboard
(ctrl-C on Windows and command-C on Mac)
then simply click the _PASTE KEY_ button in SurveyCTO Desktop.

> If the key you pasted is on the excepted format,

Reviewer: ```suggestion
If the key you pasted is on the expected format,
```

Maybe a typo for "accepted"? But I think this is better anyway

then you will see a green checkmark
and the text "_Private key successfully pasted_".
However, SurveyCTO has not tested if you pasted the correct key,
it will only test that when you are actually downloading data.
If you have pasted the incorrect key, then you will get an error
and you will not be able to read the data.

<img src="https://github.com/worldbank/dime-standards/blob/scto-guidelines/dime-research-standards/pillar-4-data-security/data-security-resources/img/scto-sync-1.png" width="75%"><!--- Image is read from master branch or use full URL-->

> ## Topic 5 - Testing your setup before

Reviewer: before what???? ;-)

Author: done in efac4632a617c269d6083b8a7ffa1a2ff104ea00

Author: done in efac4632a617c269d6083b8a7ffa1a2ff104ea00

Testing your encryption and decryption workflow is easy
but we cannot stress enough how important it is that
you do indeed test the workflow before you start to collect real data.

The first thing you want to make sure that your form is indeed encrypted.
The easiest way to do that is to log in to your server and go to the _Design_ tab.

> Find your survey and see if the icon for encryption is an open or closed pad lock.

Reviewer: ```suggestion
Find your survey and see if the icon for encryption is an open or closed padlock.
```

See the examples below where the form named _encrypted_form_ has a closed padlock
and the form named _unencrypted_form_ has an open padlock.
IF the padlock is closed then you know that all fields that
are **not** listed as publishable will be encrypted during data collection.

<img src="https://github.com/worldbank/dime-standards/blob/scto-guidelines/dime-research-standards/pillar-4-data-security/data-security-resources/img/scto-test-1.png" width="100%"><!--- Image is read from master branch or use full URL-->

> However, there is a second things you really want to test,

Reviewer: ```suggestion
However, there is a second thing you really want to test,
```

and that is to make sure you are able to decrypt the encrypted data.
To test your that you can decrypt data,

> simply submit one mock data submission,

Reviewer: ```suggestion
simply submit one trial data submission,
```

then copy the private key from your password manager,

> and use SurveyCTO Desktop to download your mock submission.

Reviewer: ```suggestion
and use SurveyCTO Desktop to download your trial submission.
```

In SurveyCTO, the public key is used at the point of data collection to encrypt your data as it is entered, 
and the private key is used at the point of downloading the data to decrypt your data. 
The server and the data entry teams will have the public key provided to them through the survey software automatically. 
The public key cannot decrypt the data.
Each public/private key pair is unique and no other private key can decrypt the data you encrypted with your public key.
You should never use the same key pair for two different projects,
but you do not need to create a new key pair for every form within a project.
We think that a good balance is to use one key pair for each project, and this practice typically conforms with IRB rules for regulating data access at the project level.

As the name suggests, it is essential to keep your private key secret. They key provides access to your encrypted data.
If you accidentally publish your private key or store it in an insecure location, then your data is no longer securely protected.
If you lose access to your private key, then there is no way to decrypt your data -- neither the software nor the service provider can restore it.

> 

Author: ```suggestion
```
